rskj-core/src/main/java/org/ethereum/rpc/exception/RskErrorResolver.java

```diff
@@ -11,49 +11,54 @@
 import org.slf4j.LoggerFactory;
 
 import java.lang.reflect.Method;
+import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static javax.xml.bind.DatatypeConverter.printHexBinary;
 
 /**
  * Created by mario on 17/10/2016.
  */
 public class RskErrorResolver implements ErrorResolver {
+    private final Map<Class<? extends Throwable>, TriFunction<Throwable, Method, List<JsonNode>, JsonError>> errorHandlers;
```

Comment:
I like the idea of use the Map to store the different kind of errors but in the other hand I don't like too much the TriFunction option.
As you are refactoring the error resolution I think it would be nice to extract each exception logic from this class to an specific class. 
We can create an interface like: (any other name would be better)
`interface JsonErrorHandler{
    ErrorResolver.JsonError apply(Throwable t, Method method, List<JsonNode> arguments);
}`
Having then a map like   
`Map<Class<? extends Throwable>, JsonErrorHandler errorHandler> errorHandlers;
`
IMHO it would be cleaner, what do you think?